src/Common/tests/System/AssertExtensions.cs

```diff
@@ -24,8 +24,18 @@ public static void Throws<T>(string netCoreParamName, string netFxParamName, Act
             string expectedParamName =
                 RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework") ?
                 netFxParamName : netCoreParamName;
-            
-            Assert.Equal(expectedParamName, exception.ParamName);
+
+            if (!RuntimeInformation.FrameworkDescription.StartsWith(".NET Native"))
```

Comment:
can you put IsNetNative in PlatformDetection.cs?

We already have this
```c#
       public static bool IsFullFramework => RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework", StringComparison.OrdinalIgnoreCase);
```
and the code here should be using that.